### DISCUSSION TITLE
Not working on my machine

### DISCUSSION THREAD

**Discussion by stephenhauck** [unoplatform/uno#18192] [General]
"/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj" (Restore target) (1) ->
  /Users/stephenhauck/.nuget/packages/uno.sdk/5.3.90/targets/Uno.ProjectCapabilities.targets(2,17): error MSB4113: Specified condition "!$(IsWinAppSdk)" evaluates to "!" instead of a boolean. [/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj] 

[msbuild.binlog.zip](https://github.com/user-attachments/files/16982050/msbuild.binlog.zip)

**Comment by morning4coffe-dev** [comment]
Hey @stephenhauck,

if I am not mistaken, @kazo0 is currently looking into this issue.

**Reply by jeromelaban** [reply]
Thanks for the report @stephenhauck. It looks like you're building with the `msbuild` command, and not using `dotnet build`. If you're using vscode, building can be done by choosing and launching the appropriate debugger.

**Reply by stephenhauck** [reply]
Thanks but I tried to debug on my Mac and that's what didn't work ...
It shouldn't be this hard  .... I will check back in Q1 2025 and see where things are at ...
I should be able to run the setups and then when it says everything is ok it should simply work ....

**Reply by jeromelaban** [reply]
It's definitely not this hard. What did you try? Can you provide specific logs? 

Also know that at this time, though, there's a specific VS Code issue with C# Dev Kit (https://github.com/unoplatform/uno/issues/18139) which requires to revert to a specific version (1.9.55) to make it work.

**Reply by stephenhauck** [reply]
Thanks for your concern .. I believe you are trying to make a good product .. and I would love to be able to use it 
Every time I have looked at the tool it wants me to use a specific version of this assembly and it doesn't use it or resolve it by default and now things like this "revert for C# devkit" ... I know it's not you guys but since it doesn't pay my bills it simpler to stay away until it matures more ... or just keep being forced to use Flutter .... 
I even subscribed to Rider in hopes the tool would be useable and I could build my apps with it ... I have since canceled that subscription .... 


I opened the project like the online help says to 
<img width="1057" alt="image" src="https://github.com/user-attachments/assets/efc60aa2-f0e7-4045-befc-a2e6e459b003">

Then I tried to debug it ...
<img width="412" alt="image" src="https://github.com/user-attachments/assets/3b076d9d-ee8d-4351-a940-ada2ad898dad">

That's where the wheels fall off ...

**Reply by jeromelaban** [reply]
Thanks. When you click on the debug button, what happens? Can you show some logs?

**Reply by stephenhauck** [reply]
I already shared the logs .. not sure what else you want ?

[{
	"resource": "/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/Uno.Resizetizer.targets",
	"owner": "msCompile",
	"code": "MSB3021",
	"severity": 8,
	"message": "Unable to copy file \"/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/runtimes/osx/native/libHarfBuzzSharp.dylib\" to \"/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/libHarfBuzzSharp.dylib\". Access to the path '/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/libHarfBuzzSharp.dylib' is denied. [/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj::TargetFramework=net8.0-desktop]",
	"source": "cpp",
	"startLineNumber": 669,
	"startColumn": 3,
	"endLineNumber": 669,
	"endColumn": 3
},{
	"resource": "/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/Uno.Resizetizer.targets",
	"owner": "msCompile",
	"code": "MSB3021",
	"severity": 4,
	"message": "Unable to copy file \"/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/runtimes/osx/native/libHarfBuzzSharp.dylib\" to \"/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/libHarfBuzzSharp.dylib\". Access to the path '/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/libHarfBuzzSharp.dylib' is denied. [/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj::TargetFramework=net8.0-desktop]",
	"source": "cpp",
	"startLineNumber": 659,
	"startColumn": 3,
	"endLineNumber": 659,
	"endColumn": 3
}]




 *  Executing task: dotnet build /Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj /property:GenerateFullPaths=true /property:TargetFramework=net8.0-desktop /consoleloggerparameters:NoSummary 

  Determining projects to restore...
  All projects are up-to-date for restore.
/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/Uno.Resizetizer.targets(659,3): warning MSB3021: Unable to copy file "/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/runtimes/osx/native/libHarfBuzzSharp.dylib" to "/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/libHarfBuzzSharp.dylib". Access to the path '/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/libHarfBuzzSharp.dylib' is denied. [/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj::TargetFramework=net8.0-desktop]
/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/Uno.Resizetizer.targets(659,3): warning MSB3021: Unable to copy file "/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/runtimes/osx/native/libSkiaSharp.dylib" to "/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/libSkiaSharp.dylib". Access to the path '/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/libSkiaSharp.dylib' is denied. [/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj::TargetFramework=net8.0-desktop]
/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/Uno.Resizetizer.targets(669,3): error MSB3021: Unable to copy file "/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/runtimes/osx/native/libHarfBuzzSharp.dylib" to "/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/libHarfBuzzSharp.dylib". Access to the path '/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/libHarfBuzzSharp.dylib' is denied. [/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj::TargetFramework=net8.0-desktop]
/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/Uno.Resizetizer.targets(669,3): error MSB3021: Unable to copy file "/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/runtimes/osx/native/libSkiaSharp.dylib" to "/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/libSkiaSharp.dylib". Access to the path '/Users/stephenhauck/.nuget/packages/uno.resizetizer/1.5.2/build/netstandard2.0/libSkiaSharp.dylib' is denied. [/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj::TargetFramework=net8.0-desktop]

 *  The terminal process "dotnet 'build', '/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj', '/property:GenerateFullPaths=true', '/property:TargetFramework=net8.0-desktop', '/consoleloggerparameters:NoSummary'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it.

**Reply by jeromelaban** [reply]
Thanks. This is not the same log, this one is also very useful.

It looks like you're hitting a race condition bug (https://github.com/unoplatform/uno/issues/17897) that we've fixed yesterday, we'll be pushing an update very soon. In the meantime, you could try to delete the `/Users/stephenhauck/.nuget/packages/uno.resizetizer` path and try again. Building twice may also help.

If you're willing to help us out further, you can also try to update to the dev build of Uno.Resizetizer, by adding this to the `Directory.Build.props` file:

```xml
<PropertyGroup>
  <UnoResizetizerVersion>1.6.0-dev.26</UnoResizetizerVersion>
</PropertyGroup>
```

Then rebuild to see if it helps. If it fails, can you post the vs logs again?

Thanks!

**Reply by stephenhauck** [reply]
I'm guessing that there needs to be an additional NuGet source added ?

**Reply by jeromelaban** [reply]
No, it's exactly this block to add in the csproj. The Uno SDK handles references automatically.

**Reply by stephenhauck** [reply]
I deleted this uno.resizetizer
I put it into the Directory.Build.props file and ran this in terminal according to your docs
msbuild /r /bl SimpleCalculator.csproj

stephenhauck@Stephens-MacBook-Air SimpleCalculator % msbuild /r /bl SimpleCalculator.csproj
Microsoft (R) Build Engine version 16.10.1 for Mono
Copyright (C) Microsoft Corporation. All rights reserved.

/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/15.0/bin/MSBuild.dll /bl /r SimpleCalculator.csproj
Build started 9/13/2024 4:18:58 PM.
Project "/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj" on node 1 (Restore target(s)).
/Users/stephenhauck/.nuget/packages/uno.sdk/5.3.90/targets/Uno.ProjectCapabilities.targets(2,17): error MSB4113: Specified condition "!$(IsWinAppSdk)" evaluates to "!" instead of a boolean. [/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj]
Done Building Project "/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj" (Restore target(s)) -- FAILED.
Deferred Messages
  
  Detailed Build Summary
  ======================
      
  
  ============================== Build Hierarchy (IDs represent configurations) =====================================================
  Id                  : Exclusive Time   Total Time   Path (Targets)
  -----------------------------------------------------------------------------------------------------------------------------------
  0                   : 1.141s           1.141s       /Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj (Restore) 
  
  ============================== Node Utilization (IDs represent configurations) ====================================================
  Timestamp:            1        Duration   Cumulative
  -----------------------------------------------------------------------------------------------------------------------------------
  638618555388492320:   0        1.157s     1.157s .......................
  -----------------------------------------------------------------------------------------------------------------------------------
  Utilization:          100.0    Average Utilization: 100.0

Build FAILED.

"/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj" (Restore target) (1) ->
  /Users/stephenhauck/.nuget/packages/uno.sdk/5.3.90/targets/Uno.ProjectCapabilities.targets(2,17): error MSB4113: Specified condition "!$(IsWinAppSdk)" evaluates to "!" instead of a boolean. [/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.37
stephenhauck@Stephens-MacBook-Air SimpleCalculator %

**Reply by jeromelaban** [reply]
`msbuild` is not a supported command for since .NET 6 and must not be used anymore. It was only for Xamarin targets, and is likely still on your system because VS for Mac is installed.

Make sure to use `dotnet build`. Our vscode uses those appropriate commands automatically.

**Reply by stephenhauck** [reply]
stephenhauck@Stephens-MacBook-Air SimpleCalculator % dotnet build  /bl SimpleCalculator.csproj 
  Determining projects to restore...
/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj : error NU1102: Unable to find package Uno.Resizetizer with version (>= 1.6.0-dev.26)
/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj : error NU1102:   - Found 135 version(s) in NuGet official package source [ Nearest version: 1.6.0-dev.22 ]
/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj : error NU1102:   - Found 5 version(s) in uno dev [ Nearest version: 1.5.2 ]
/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj : error NU1102:   - Found 0 version(s) in /usr/local/share/dotnet/library-packs
  Failed to restore /Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj (in 3.37 sec).

Build FAILED.

/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj : error NU1102: Unable to find package Uno.Resizetizer with version (>= 1.6.0-dev.26)
/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj : error NU1102:   - Found 135 version(s) in NuGet official package source [ Nearest version: 1.6.0-dev.22 ]
/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj : error NU1102:   - Found 5 version(s) in uno dev [ Nearest version: 1.5.2 ]
/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj : error NU1102:   - Found 0 version(s) in /usr/local/share/dotnet/library-packs
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:05.06
stephenhauck@Stephens-MacBook-Air SimpleCalculator % 

[msbuild.binlog.zip](https://github.com/user-attachments/files/16999594/msbuild.binlog.zip)

**Reply by jeromelaban** [reply]
Thanks. Your machine is not detecting this version of the package: https://www.nuget.org/packages/Uno.Resizetizer/1.6.0-dev.26. You may want to try again, it may be a caching issue.

**Comment by stephenhauck** [comment]
hitting run 
<img width="487" alt="image" src="https://github.com/user-attachments/assets/f3036e9b-75e4-4eec-8c61-895a17bd3aff">
<img width="1070" alt="image" src="https://github.com/user-attachments/assets/61aab589-4604-4461-8b6d-ca5fd644b7e5">



generates this ...

  *  Executing task: dotnet build /Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj /property:GenerateFullPaths=true /property:TargetFramework=net8.0-desktop /consoleloggerparameters:NoSummary 

  Determining projects to restore...
/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj : error NU1102: Unable to find package Uno.Resizetizer with version (>= 1.6.0-dev.26)
/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj : error NU1102:   - Found 135 version(s) in NuGet official package source [ Nearest version: 1.6.0-dev.22 ]
/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj : error NU1102:   - Found 5 version(s) in uno dev [ Nearest version: 1.5.2 ]
/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj : error NU1102:   - Found 0 version(s) in /usr/local/share/dotnet/library-packs
  Failed to restore /Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj (in 2.41 sec).

 *  The terminal process "dotnet 'build', '/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj', '/property:GenerateFullPaths=true', '/property:TargetFramework=net8.0-desktop', '/consoleloggerparameters:NoSummary'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it. 

 *  Executing task: dotnet build /Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj /property:GenerateFullPaths=true /property:TargetFramework=net8.0-desktop /consoleloggerparameters:NoSummary 

  Determining projects to restore...
/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj : error NU1102: Unable to find package Uno.Resizetizer with version (>= 1.6.0-dev.26)
/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj : error NU1102:   - Found 135 version(s) in NuGet official package source [ Nearest version: 1.6.0-dev.22 ]
/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj : error NU1102:   - Found 5 version(s) in uno dev [ Nearest version: 1.5.2 ]
/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj : error NU1102:   - Found 0 version(s) in /usr/local/share/dotnet/library-packs
  Failed to restore /Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj (in 2.64 sec).

 *  The terminal process "dotnet 'build', '/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj', '/property:GenerateFullPaths=true', '/property:TargetFramework=net8.0-desktop', '/consoleloggerparameters:NoSummary'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it.

**Comment by stephenhauck** [comment]
Still no dice .. will check back later .. it shouldn't be this hard to install and run samples when uno check is sucessful.
Thanks for your efforts ...

**Comment by stephenhauck** [comment]
I pulled latest for the samples and there were only some minor changes to the source but I was able to run on my Mac .. 
The I tried to run on an Android emulator and get this error 
<img width="765" alt="image" src="https://github.com/user-attachments/assets/b99362f9-5600-4134-ba87-d58096c0725f">

Seems it is trying to run / execute on the filesystem outside of my profile .... not sure why ?

**Reply by jeromelaban** [reply]
Thanks! You've gotten further indeed. What's the exception text/description that you're getting?

**Reply by jeromelaban** [reply]
> It’s in the attached screenshot …. It’s trying to execute outside of my profile and Mac’s don’t allow that … Uno is the only tool that does this … no idea why …. Rider didn’t and Flutter doesn’t …

Thanks, but no, there is more information available, most likely in another window or the app's log window in Rider. This is the stack trace to the exception, not the exception itself. Can you try uninstalling the app from the device/emulator, and remove the bin/obj folders then retry the build?

**Reply by stephenhauck** [reply]
I will look again later but I'm using VS Code and not Rider, I did not have this problem in Rider.
The error is that the app / command being executed is outside of my user profile and does not have authority 
Delteing the bin / obj or app from the device (emulator) wont matter because whatever is being passed to the command line is not including the profile path 

I'm sure you know this but on a Mac you cannot execute scripts outside of your profile and such unless you use sudo ... 

<img width="237" alt="image" src="https://github.com/user-attachments/assets/ad61f59e-0142-4dc9-9788-f94dedeae13d">

**Reply by jeromelaban** [reply]
I'm not sure of what you mean by paths outside of your profile, but if you're referring to "/User/runner/..." those are .NET sources symbols paths that are unrelated to anything the app does. It's only used to show sources when clicking on stack frames.

Please try uninstalling the app and deleting the bin/obj folders of your source tree before rebuilding. Thanks!

**Reply by stephenhauck** [reply]
I deleted the bin and obj from the source directory and hit run 

Android Emulator 'pixel_5_-_api_29' already running...
Unable to find the app's APK. (The path '/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/bin\Debug/net8.0-android/android-arm64/' does not exist)

<img width="1128" alt="image" src="https://github.com/user-attachments/assets/b386b8f5-0805-43ca-85ac-59901527a31c">

**Reply by jeromelaban** [reply]
Thanks for the report. There's a binlog in the project's folder, could you provide it? It seems like the build did not finish successfully to create an APK.

**Reply by stephenhauck** [reply]
[msbuild.binlog.zip](https://github.com/user-attachments/files/17118269/msbuild.binlog.zip)

**Reply by jeromelaban** [reply]
Thanks. It seems like this time, you're hitting a .NET Android race condition of some kind:

```
/usr/local/share/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.138/tools/Xamarin.Android.EmbeddedResource.targets(39,5): error XARLP7000: Xamarin.Tools.Zip.ZipException: Renaming temporary file failed: No such file or directory
   at Xamarin.Tools.Zip.ZipArchive.Close() in /Users/runner/work/1/s/LibZipSharp/Xamarin.Tools.Zip/ZipArchive.cs:line 989
   at Xamarin.Android.Tasks.ZipArchiveEx.Dispose(Boolean disposing)
   at Xamarin.Android.Tasks.ZipArchiveEx.Dispose()
   at Xamarin.Android.Tasks.ResolveLibraryProjectImports.<>c__DisplayClass75_0.<CreateResourceArchive>b__0(String f)
   at Microsoft.Android.Build.Tasks.Files.ArchiveZipUpdate(String target, Action`1 archiver) in /Users/runner/work/1/s/xamarin-android/external/xamarin-android-tools/src/Microsoft.Android.Build.BaseTasks/Files.cs:line 92
   at Xamarin.Android.Tasks.ResolveLibraryProjectImports.CreateResourceArchive(String resDir, String outputFile)
   at Xamarin.Android.Tasks.ResolveLibraryProjectImports.Extract(IDictionary`2 jars, ICollection`1 resolvedResourceDirectories, ICollection`1 resolvedAssetDirectories, ICollection`1 resolvedEnvironments, ICollection`1 proguardConfigFiles)
   at Xamarin.Android.Tasks.ResolveLibraryProjectImports.RunTask()
   at Microsoft.Android.Build.Tasks.AndroidTask.Execute() in /Users/runner/work/1/s/xamarin-android/external/xamarin-android-tools/src/Microsoft.Android.Build.BaseTasks/AndroidTask.cs:line 25 [/Users/stephenhauck/Documents/GitHub/Uno.Samples/reference/SimpleCalc/CSharp-MVUX/SimpleCalculator/SimpleCalculator.csproj]
```

Which explains why the APK was not generated. I would not be surprised that re-starting the build would make it work. I've opened an issue on the .NET Android side we can keep track: https://github.com/dotnet/android/issues/9321.

**Reply by stephenhauck** [reply]
Nope .. tried that before and tried it again just now ....
Uno splash shows then crash 
<img width="904" alt="image" src="https://github.com/user-attachments/assets/9d501b29-eec9-43da-95b6-073b425233db">

I then did a clean boot of the emulator with no apps on it and it finally worked ......
Seems something is not cleaning up on deploy ????

**Reply by jeromelaban** [reply]
This definitely looks like a missing caching likely caused by an unfinished build. By the last comment on the issue I opened, this may be a `ulimit` type of issue on your system.

Can you provide the output of `ulimit -n` and `ulimit -u` from a terminal?

**Reply by stephenhauck** [reply]
<img width="591" alt="image" src="https://github.com/user-attachments/assets/0ba7bb67-ce7b-4fad-b962-1bed8f35fcf9">

**Reply by jeromelaban** [reply]
Thanks. You may want to try setting the files limit with `ulimit -n 4096` and see if it helps. This setting is not persisted, and if it helps, you may need to set it to a more central location depending on your version of macOS.

**Reply by stephenhauck** [reply]
not sure why this is suddenly necessary .. never had to do this for anything else .NET or Flutter or Swift ......

**Reply by jeromelaban** [reply]
Every platform is different. In this case, it may be that .NET for vscode in this context opens more files that is regularly needed and it is causing issues on your system.

**Comment by stephenhauck** [comment]
It’s in the attached screenshot …. It’s trying to execute outside of my profile and Mac’s don’t allow that … Uno is the only tool that does this … no idea why …. Rider didn’t and Flutter doesn’t ….On Sep 23, 2024, at 9:03 AM, Jérôme Laban ***@***.***> wrote:﻿
Thanks! You've gotten further indeed. What's the exception text/description that you're getting?
